core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1478,6 +1484,12 @@ public boolean exists(AlluxioURI path, ExistsContext context)
           createLockingScheme(path, context.getOptions().getCommonOptions(), LockPattern.READ),
           rpcContext.getJournalContext())
       ) {
+        try {
```

Comment:
what is the reason to check two times?

Reply:
Permissions should be checked before and after metadata synchronization.

Reply:
Given the permission is checked from no permission to permission in the UFS, this will fail the sync, the user doesn't have access to the file. And the user will be allowed to access only others triggers the sync.

Is it right?

Reply:
Ignore FileDoesNotExistException and check permission after metadata sync.